### ISSUE TITLE
m.request url parameters used in simple app tutorial

### ISSUE THREAD

**Issue by spacejack** [lhorie/mithril.js#1788]
In the simple app tutorial, the load user request url is built like this:

```javascript
m.request({
  method: "GET",
  url: "https://rem-rest-api.herokuapp.com/api/users/:id",
  data: {id: 1},
  withCredentials: true,
})
```

But this results in the url: https://rem-rest-api.herokuapp.com/api/users/1?id=1

It doesn't break the tutorial app (the query string is ignored by the server) but it's not quite right. Should the tutorial be updated or is this a problem with `m.request`?

[jsbin example](http://jsbin.com/tuyowazine/1/edit?js,output)

**Comment by pygy** [comment]
URL `:arguments` interpolation is problematic in other areas as well, I think m.requests tries to do too many things with the `data` field.

We could have `interpolate`, and `payload` fields, where payload is passed as a query string for GET and as body for POST, or even distinct options...

**Comment by spacejack** [comment]
Is the intention for adding a `payload` field to preserve backwards-compatibility with `data`? Otherwise my intuition would be to have data remain used as-is for POST body and GET query strings (similar to jQuery), then add a `urlParams` property for interpolation.

In which case I suppose you could fall back on old behaviour when `urlParams` is omitted?

Regarding the name, I'm not too picky, but `interpolate` sounds like a boolean to me while something like `params` sounds more like an object, and `urlParams` tells you exactly what it will be used for.

**Comment by pygy** [comment]
> Is the intention for adding a `payload` field to preserve backwards-compatibility with `data`?

Yes, we could even go as far as `urlParams`, `queryString` and `body`. `data` would be ignored if any of these were specified (and marked as deprecated in the docs).

Agreed on `interpolate`.

@orbitbot @fibrefox and @isiahmeadows have discussed similar issues at length in #1699, but I've skipped that discussion so far.

**Comment by spacejack** [comment]
Ok, in the meantime maybe I'll submit a PR for the tutorial to prevent the extraneous querystring.
